File: dcp-solutions/2025-01-25/solution.php
```php
class BitArray {
    private $size;
    private $array;

    public function __construct($size) {
        $this->size = $size;
        $this->array = array_fill(0, $size + 1, 0);
    }

    public function set($i, $val) {
        if ($i < 0 || $i >= $this->size) {
            throw new Exception("Index out of range");
        }
        if (!in_array($val, [0, 1])) {
            throw new Exception("Invalid value. Must be either 0 or 1");
        }
        $i++;
        while ($i <= $this->size) {
            $this->array[$i] = $val;
            $i += $i & -$i;
        }
    }

    public function get($i) {
        if ($i < 0 || $i >= $this->size) {
            throw new Exception("Index out of range");
        }
        $i++;
        $result = 0;
        while ($i > 0) {
            $result += $this->array[$i];
            $i -= $i & -$i;
        }
        return $result;
    }
}
```

File: dcp-solutions/2025-01-25/solution.c
```c
#include <stdio.h>

typedef struct {
    int* bits;
    int size;
} BitArray;

BitArray* init(int size) {
    BitArray* bitArray = (BitArray*) malloc(sizeof(BitArray));
    if (!bitArray) {
        return NULL;
    }
    
    bitArray->bits = (int*) calloc(size, sizeof(int));
    if (!bitArray->bits) {
        free(bitArray);
        return NULL;
    }
    bitArray->size = size;
    
    for (int i = 0; i < size; ++i) {
        bitArray->bits[i] = 0;
    }
    
    return bitArray;
}

void set(BitArray* bitArray, int index, int val) {
    if (val == 1 || val == 0) {
        bitArray->bits[index] = val;
    } else {
        printf("Invalid value. Value should be either 0 or 1.\n");
    }
}

int get(BitArray* bitArray, int index) {
    return bitArray->bits[index];
}
```